### ISSUE TITLE
Have you tried to set densenet as base network?

### ISSUE THREAD

**Issue by evercherish** [sfzhang15/RefineDet#18]
Densenet may score a higher mAP compared to VGG and Res101.
If you have finished this, would you mind sharing a Densenet_VOC2007_320.py?
thanks!

**Comment by sfzhang15** [comment]
@evercherish Hi, sorry, until now, we just use VGG16 and Res101 as our base network. In the future to expand this work, we have the plan to try more base networks, such as MobileNet, ShuffleNet, DenseNet and so on. If we finish, we will share them.

**Comment by evercherish** [comment]
looking forward to it
Thanks

**Comment by lengly** [comment]
@sfzhang15 @evercherish 
I tried to use DenseNet as base network.
I use the DenseNet pre-trained model from https://github.com/shicai/DenseNet-Caffe
This is the only public pre-trained caffe model I found. But the main problem is the growth rate in that model is 32. Do you know what does this means? It means we need extremely huge memory... When image-size=512, we can only set batch-size=4 on V100x4 GPU(each memory is 16GB)...

Actually we don't need set growth rate so big, but I didn't find other pre-trained models...

So if you want to use DenseNet, my suggestion is to set growth rate equals to 12, and train this DenseNet on ImageNet by yourself, then use your own pre-trained model...

**Comment by sfzhang15** [comment]
@lengly Thanks for sharing. According to your experience, the pre-trained DenseNet model (i.e., K=32) takes more GPU memory than ResNet101. For ResNet101, a 24G GPU only can be input five 512x512 images and the BN layer is barely stable. If there are less than 5 images in one GPU, the BN layer will be unstable. So the pre-trained DenseNet model (i.e., K=32) is computationally prohibitive even for a 24G GPU.

**Comment by lengly** [comment]
@sfzhang15 When you train ResNet101, did you notice some iteration is really slow? When image size=320, batch=32, it takes 1 second per iteration normally, but sometimes, it takes 5x or even 10x slower  (10 seconds per iter)

**Comment by sfzhang15** [comment]
@lengly Yes, we also have this phenomenon, but it rarely appears. Maybe when other users on the server take up CPU or IO resources, or when the server is busy doing other tasks, this phenomenon will appear.

**Comment by evercherish** [comment]
@lengly Could you please share your  python script? 
thanks!

**Comment by lengly** [comment]
@evercherish   Sorry, our company didn't allow us to do this...

**Comment by evercherish** [comment]
@lengly That's all right. Thanks you for your reply.
